client/js/controllers/editor.js

```diff
@@ -831,6 +831,74 @@
+                            cached.pop();
+                            file = cached.join("/");
+                        }
+                        if($scope.folder !== 'demo') {
+                        var dialog = $('<div></div>').
+                                html($compile('<form ng-submit="Upload()"><input' +
+                                            ' type="file" id="inputFile" class="inputFileControls"' +
+                                            ' onChange="angular.element(this).scope().filesChanged(this)"/>'+
+                                            ' <label for="inputFile">Select Files</label>'+
+                                            ' <li class="list-item" ng-repeat="file in files">{{file.name}}</li>'+
+                                            ' </form>')($scope)).
+                        dialog({
+                          title: "Choose file to import",
+                          autoOpen: false,
+                          modal: true,
+                          position: { at: "center top"},
+                          height: 225,
+                          width: 300,
+                          show: { effect: "fade", duration: 300 },
+                          hide: { effect: "fade", duration: 300 },
+                          resizable: 'disable',
+                          buttons: {
+                              "Upload": function() {
+                                  var formdata = new FormData();
+                                  var data=String(file);
+                                  formdata.append("upload_path", data);
+                                  angular.forEach($scope.files,function(file) {
+                                  formdata.append('file', file)
+                                  });
+                                  $http.post('/api/file/upload',formdata, {
+                                    transformRequest: angular.identity,
+                                    headers:{'Content-Type':undefined}
+                                  })
+                                    .success(function(d) {
+                                        console.log(d)
+                                        $scope.refreshTree();                                        
+                                    });
+                                  $(this).dialog("close");
+                              },
+                              Cancel: function() {
+                                  $(this).dialog("close");
+                                  }
+                              },
+                              close: function(ev, ui) {
+                                  $(this).dialog("close");
+                              }
+                      });
+                      dialog.dialog("open");
+                        }else{
+                            console.log("Error: Upload to demo folder not allowed");
+                            alert("Error: Upload to demo folder Forbidden!")
```

Comment:
To me the best message here would be:
"Error: Upload to demo folder is not allowed"

Also, you can remove the console.log here since we already have the alert